aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java

```diff
@@ -85,70 +84,32 @@ private AppSyncRequestFactory() {}
      * @throws DataStoreException On Failure to inspect
      */
     @NonNull
-    static <T extends Model> String buildSyncDoc(
-            @NonNull final Class<T> modelClass,
+    static <T, M extends Model> AppSyncGraphQLRequest<T> buildSyncRequest(
```

Comment:
What are the possible resolutions for `T`, here -- `Iterable<String>`, and importantly, some upcoming type of paginated result?

Reply:
Yep, right now it is `Iterable<String>`, but my next PR will change it to be `PaginatedResult<ModelWithMetadata<M>>`